lib/pbench/test/unit/server/test_unpack_tarballs.py

```diff
@@ -0,0 +1,117 @@
+import glob
+from pathlib import Path
+import pytest
+
+from pbench.common.logger import get_pbench_logger
+from pbench.server.filetree import FileTree, TarballUnpackError
+from pbench.server.unpack_tarballs import Unpack
+
+
+@pytest.fixture()
+def make_logger(server_config):
+    """
+    Construct a Pbench Logger object
+    """
+    return get_pbench_logger("TEST", server_config)
+
+
+class TestUnpackTarballs:
+    tar_name = "A.tar.xz"
+    tarlist = [tar_name]
+    linksrc = "TO-UNPACK"
+
+    def mock_glob(self, path):
+        """Returns the Tarball list"""
+        return TestUnpackTarballs.tarlist
+
+    def mock_resolve(self: Path, strict: bool = False):
+        """
+        Return a "full" fake file name
+
+        Args:
+            strict: Not used in the mock
+
+        Returns:
+            new Path with fake resolution
+        """
+        return Path(self.tar_name)
+
+    def test_tarball_symlinks(self, monkeypatch, server_config, make_logger, caplog):
+        """Tests for an incorrect symlink which results in an unresolved location"""
+
+        monkeypatch.setattr(glob, "iglob", self.mock_glob)
+        obj = Unpack(server_config, self.linksrc, make_logger)
+        total, success = obj.unpack_tarball()
```

Comment:
I can't tell what the "unit" is which this scenario is testing.  Apparently, it's testing the stack all the way down to the file system.  That is, you're mocking `glob.iglob()` but not `Path.resolve()`, so we're actually going to do a file system operation, hoping that the target file doesn't just happen to exist.

I'm willing to accept that the "unit" is all of the Pbench stack, including `FileTree` and `get_tarball_md5()` (although, I think it would be better to test `Unpack.unpack_tarball()` in isolation), but I think you should be mocking all external dependencies, in particular, file system ones.  So, this test should have a mock for `Path.resolve()` which raises `FileNotFoundError` (and, sadly, you cannot do that in a lambda expression, so I would use an inner-function for that).

Reply:
What about using this expresiion? `lambda x: exec('raise(Exception(x))')`

Reply:
To be honest, I don't know whether I like that more than I hate it.  😆 

I like the fact that it is compact and relatively simple, especially because that allows the code to be placed near where the reader needs it; however, for no reason that I can name, I'm opposed to using `exec()` except in very rare circumstances (of which this isn't one...).

So, I'll leave it up to you (and anyone else who has an opinion).

Reply:
Having looked at the latest update to the code, I think having the mock be a function instead of a lambda expression would be best, because, in addition to raising an exception, I think it should be asserting that `strict` is true, and, while you _could_ do that in the `exec()` call as well, that makes it complicated enough that I think we should just use a function.